bitswap/client/internal/peermanager/peerwantmanager.go

```diff
@@ -16,6 +19,31 @@ type Gauge interface {
 	Dec()
 }
 
+// BroadcastConfig contains broadcast reduction settings. The zero-value is the
+// default configuration and results in the minimum amount of broadcasts
+// without placing and hard limit on the number of broadcasts.
+type BroadcastConfig struct {
+	// LocalAlways is a Peerstore that is uses to determine if a peer is on the
+	// local network. If non nil, always boradcast to peers on the local
+	// network. If nil, apply broadcast reduction logic to peers on the local
+	// network the same as peers on other networks.
+	LocalAlways peerstore.Peerstore
```

Comment:
I don't get the meaning of `LocalAlways`, why not something like `Peerstore`/`LocalPeerstore`?

Reply:
It is a way to combine a boolean flag (`LocalAlways`) with the `Peerstore` that is used to remember peers that are on the local network. If the  `Peerstore` is not `nil` then this means that `LocalAlways` flag is true.

I could change this to a flag and an optional peerstore, but there there would need to be an additional check if the flag is `true` then the peerstore must be provided. WDYT?